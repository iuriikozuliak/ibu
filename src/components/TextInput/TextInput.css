@value dark from 'theme/colors.css';
@value unit from 'theme/size.css';
@value danger from 'theme/colors.css';

.TextInput {
  border: 1px solid dark;
  outline: none;
  height: 50px;
  padding: 0 20px;
  width: 100%;
  font-size: 16px;
  font-weight: 400;
}

.TextInputError {
  color: danger;
  font-size: .75em;
  position: absolute;
  padding-top: unit;
}
