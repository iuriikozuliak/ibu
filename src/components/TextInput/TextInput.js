import React     from 'react';
import PropTypes from 'prop-types';

import styles from './TextInput.css';

const TextInput = ({
  placeholder
}) => (
  <input
    type        = "text"
    className   = { styles.TextInput }
    placeholder = { placeholder } />
);

TextInput.defaultProps = {
  placeholder: ''
};

TextInput.propTypes = {
  placeholder: PropTypes.String
};

export default TextInput;
