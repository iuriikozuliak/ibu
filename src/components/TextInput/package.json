{
  "name": "TextInput",
  "private": true,
  "main": "./TextInput.js"
}
