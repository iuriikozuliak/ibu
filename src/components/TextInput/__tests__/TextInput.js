import React     from 'react';
import renderer  from 'react-test-renderer';
import TextInput from '../TextInput';

test('Renders TextInput component', () => {
  const component = renderer.create(
    <TextInput />
  ).toJSON();
  expect(component).toMatchSnapshot();
});

test('Renders TextInput component with passed placeholder prop', () => {
  const component = renderer.create(
    <TextInput placeholder = "First Name" />
  ).toJSON();
  expect(component).toMatchSnapshot();
});
