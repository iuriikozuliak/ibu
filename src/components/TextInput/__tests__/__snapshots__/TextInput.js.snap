// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders TextInput component 1`] = `
<div
  className="Box"
>
  <input
    className="TextInput"
    name={undefined}
    onChange={undefined}
    placeholder=""
    type="text"
    value=""
  />
  
</div>
`;

exports[`Renders TextInput component with passed placeholder prop 1`] = `
<div
  className="Box"
>
  <input
    className="TextInput"
    name={undefined}
    onChange={undefined}
    placeholder="First Name"
    type="text"
    value=""
  />
  
</div>
`;
